### PR TITLE
Load tests

### PR DESCRIPTION
**Pull request by zmoore-usgs**
Added load tests. These aren't designed to be run as part of the normal integration tests, but figured I'd put them in the same project since they utilize the same process and harness. They do take a very long time to complete (at least on my local machine), well over an hour if they're all run.

This only covers Add Validations and full Add Transactions at the moment. Most of the reason for that is the other workflows in MLR are more or less contained within the Add workflows, so the test inherently covers most of the same logic. Additionally, doing Update and Export tests would depend on the database first having known-good data loaded into it or all of the updates would be considered invalid since the sites won't exist. That could be done but could take a fair amount of effort to get setup and I think we've gotten the most meaningful results from the tests that this does include.

### CODE REVIEW

File: reset.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (zmoore-usgs) [inline reset.sh]: This file isn't really necessary, just created it out of convenience to easily reset the state of things when necessary. Can be removed if we don't want it as part of the repo

File: tests/integrations/load/test_plan.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
Comment (isuftin) [inline tests/integrations/load/test_plan.sh]: You may want to provide a little documentation on what this does and how it is to be used. For example, what prep needs to happen before this is run? Can I just grab this project and run this script with no other prep?

Comment (zmoore-usgs) [reply tests/integrations/load/test_plan.sh]: :+1: I can add to the readme, but yeah you should be able to just pull the project and launch the services like you would for the other tests and then run the script without any other prep needed.

**Comment by isuftin** [comment]
@zmoore-usgs testing now. This seems like it's going to take a while. I wonder if you have any scripted way of reading pass/fail after the end of all the load testing

**Comment by isuftin** [comment]
I'll keep running it on my end but so far I got..

```
 tests/integrations/load/test_plan.sh
Starting load testing.
Starting load test with 3 users (Starting Site: 100000000).
Feb 13, 2019 6:35:38 PM java.util.prefs.FileSystemPreferences$1 run
INFO: Created user preferences directory.
Creating summariser <summary>
Created the tree successfully using /tests/integrations/load/load.jmx
Configuring remote engine: jmeter.server.1
Using local port: 60000
Configuring remote engine: jmeter.server.2
Configuring remote engine: jmeter.server.3
Starting remote engines
Starting the test @ Wed Feb 13 18:35:42 UTC 2019 (1550082942781)
summary +      2 in 00:00:13 =    0.2/s Avg:   309 Min:   114 Max:   505 Err:     0 (0.00%) Active: 5 Started: 2 Finished: 0
Remote engines have been started
Waiting for possible Shutdown/StopTestNow/Heapdump message on port 4445
summary +      2 in 00:01:00 =    0.0/s Avg:   571 Min:   240 Max:   903 Err:     0 (0.00%) Active: 7 Started: 24 Finished: 20
summary =      4 in 00:01:13 =    0.1/s Avg:   440 Min:   114 Max:   903 Err:     0 (0.00%)
summary +    432 in 00:13:02 =    0.6/s Avg:  6754 Min:     3 Max: 237704 Err:     0 (0.00%) Active: 9 Started: 69 Finished: 63
summary =    436 in 00:14:15 =    0.5/s Avg:  6696 Min:     3 Max: 237704 Err:     0 (0.00%)
summary +     68 in 00:00:19 =    3.6/s Avg: 65305 Min:     4 Max: 468933 Err:     0 (0.00%) Active: 0 Started: 69 Finished: 72
summary =    504 in 00:14:34 =    0.6/s Avg: 14604 Min:     3 Max: 468933 Err:     0 (0.00%)
Tidying up remote @ Wed Feb 13 18:50:23 UTC 2019 (1550083823520)
... end of run
Load test completed.
Starting load test with 12 users (Starting Site: 100003333).
Feb 13, 2019 6:50:33 PM java.util.prefs.FileSystemPreferences$1 run
INFO: Created user preferences directory.
Creating summariser <summary>
Created the tree successfully using /tests/integrations/load/load.jmx
Configuring remote engine: jmeter.server.1
Using local port: 60000
Configuring remote engine: jmeter.server.2
Configuring remote engine: jmeter.server.3
Starting remote engines
Starting the test @ Wed Feb 13 18:50:37 UTC 2019 (1550083837336)
Remote engines have been started
Waiting for possible Shutdown/StopTestNow/Heapdump message on port 4445
summary +      4 in 00:03:51 =    0.0/s Avg:   162 Min:     9 Max:   343 Err:     0 (0.00%) Active: 9 Started: 32 Finished: 24
summary +    162 in 00:00:27 =    6.0/s Avg:  3445 Min:     5 Max: 186683 Err:     0 (0.00%) Active: 9 Started: 47 Finished: 39
summary =    166 in 00:04:18 =    0.6/s Avg:  3366 Min:     5 Max: 186683 Err:     0 (0.00%)
summary +     81 in 00:00:35 =    2.3/s Avg:  5673 Min:     5 Max: 187006 Err:     0 (0.00%) Active: 7 Started: 59 Finished: 53
summary =    247 in 00:04:53 =    0.8/s Avg:  4123 Min:     5 Max: 187006 Err:     0 (0.00%)
summary +    127 in 00:00:30 =    4.3/s Avg:  7159 Min:     3 Max: 194450 Err:     0 (0.00%) Active: 8 Started: 65 Finished: 58
summary =    374 in 00:05:22 =    1.2/s Avg:  5154 Min:     3 Max: 194450 Err:     0 (0.00%)
summary +    130 in 00:07:09 =    0.3/s Avg: 33411 Min:     3 Max: 441089 Err:     0 (0.00%) Active: 0 Started: 68 Finished: 69
summary =    504 in 00:12:31 =    0.7/s Avg: 12442 Min:     3 Max: 441089 Err:     0 (0.00%)
Tidying up remote @ Wed Feb 13 19:03:13 UTC 2019 (1550084593542)
... end of run
Load test completed.
Starting load test with 24 users (Starting Site: 100016665).
Feb 13, 2019 7:03:23 PM java.util.prefs.FileSystemPreferences$1 run
INFO: Created user preferences directory.
Creating summariser <summary>
Created the tree successfully using /tests/integrations/load/load.jmx
Configuring remote engine: jmeter.server.1
Using local port: 60000
Configuring remote engine: jmeter.server.2
Configuring remote engine: jmeter.server.3
Starting remote engines
Starting the test @ Wed Feb 13 19:03:26 UTC 2019 (1550084606645)
summary +      1 in 00:00:01 =    1.4/s Avg:   103 Min:   103 Max:   103 Err:     0 (0.00%) Active: 1 Started: 1 Finished: 0
Remote engines have been started
Waiting for possible Shutdown/StopTestNow/Heapdump message on port 4445
```

I think this is good to merge

**Comment by zmoore-usgs** [comment]
Yeah, there's no scripted way of reading pass/fail at this point and it does take a very long time to run. I was just using it to generate statistics with the assumption that there'd be some failures when the number of users got too big. If we do script pass/fail checking we'd need to set some thresholds like a minimum number of users that must generate 0% errors and whatnot. Though the hardware we run on would also have an affect on the load test results

**Pull request merged**